posthog/tasks/alerts/test/test_checks.py

```diff
@@ -0,0 +1,167 @@
+import pytest
+from typing import Any, Optional
+from unittest.mock import MagicMock, patch
+
+from freezegun import freeze_time
+
+from posthog.models.instance_setting import set_instance_setting
+from posthog.tasks.alerts.checks import check_all_alerts
+from posthog.test.base import APIBaseTest, _create_event, flush_persons_and_events, ClickhouseDestroyTablesMixin
+from posthog.api.test.dashboards import DashboardAPI
+from posthog.schema import ChartDisplayType, EventsNode, TrendsQuery, TrendsFilter
+from posthog.tasks.test.utils_email_tests import mock_email_messages
+
+
+@freeze_time("2024-06-02T08:55:00.000Z")
+@patch("posthog.tasks.alerts.checks.EmailMessage")
```

Comment:
You shouldn't need to patch or mock email directly. If you follow the recommendation of making email a sepearate task of at least function, you can easily mock that to see if notification has been called.

Reply:
Yeah, makes sense, will fix it soon